lib/ansible/plugins/connections/docker.py

```diff
@@ -0,0 +1,119 @@
+# Based on the chroot connection plugin by Maykel Moya
+#
+# Connection plugin for configuring docker containers
+# (c) 2014, Lorin Hochstein
+# (c) 2015, Leendert Brouwer
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+import time
+
+from ansible import errors
+from ansible.plugins.connections import ConnectionBase
+
+BUFSIZE = 65536
+
+class Connection(ConnectionBase):
+
+    def __init__(self, play_context, new_stdin, *args, **kwargs):
+        super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
+
+        if 'docker_command' in kwargs:
+            self.docker_cmd = kwargs['docker_command']
+        else:
+            self.docker_cmd = 'docker'
+
+    @property
+    def transport(self):
+        return 'docker'
+
+    def _connect(self, port=None):
+        """ Connect to the container. Nothing to do """
+        return self
+
+    def exec_command(self, cmd, tmp_path, sudo_user=None, sudoable=False,
+                     executable='/bin/sh', in_data=None, su=None,
+                     su_user=None):
+
+        """ Run a command on the local host """
+
```

Comment:
I would call super here like this:

``` python
super(Connection, self).exec_command(cmd, tmp_path, in_data=in_data, sudoable=sudoable)
```

It doesn't do anything useful for the docker connection but once again, it's nice to do it so that anyone looking at the docker connection plugin for a basis to create their own connection plugin will do the right thing (the base class's exec_command() fetch_file(), and put_file() use the @ensure_connect decorator to make sure that our connection is open before attempting these actions.  For plugins that actually need to establish a connection, this is a necessary step).